beacon-chain/blockchain/init_sync_process_block.go

```diff
@@ -170,10 +172,23 @@ func (s *Service) generateState(ctx context.Context, startRoot [32]byte, endRoot
 	if preState == nil {
 		return nil, errors.New("finalized state does not exist in db")
 	}
-	endBlock, err := s.beaconDB.Block(ctx, endRoot)
-	if err != nil {
-		return nil, err
+
+	var endBlock *ethpb.SignedBeaconBlock
+	if featureconfig.Get().InitSyncBatchSaveBlocks && s.hasInitSyncBlock(endRoot) {
+		endBlock = s.getInitSyncBlock(endRoot)
+		if featureconfig.Get().InitSyncBatchSaveBlocks {
```

Comment:
Why is this condition here? Wouldn't it already be true?

Reply:
Oops, thanks!